lib/logmatcher.c

```diff
@@ -347,7 +347,9 @@ log_matcher_pcre_re_compile(LogMatcher *s, const gchar *re, GError **error)
     }
 
   /* optimize regexp */
-  self->extra = pcre_study(self->pattern, PCRE_STUDY_JIT_COMPILE, &errptr);
+  self->extra = pcre_study(self->pattern,
```

Comment:
could you please extract that condition (` (self->super.flags & LMF_DISABLE_JIT) ? 0 : PCRE_STUDY_JIT_COMPILE`) out of the function call and assign to a local `gint options` variable instead ? also `LMF_DISABLE_JIT` should only negate the proper bit in the options and not set to `0`.

Reply:
I extracted some functions in the area, can you check if this is ok?